src/naive.rs

```diff
@@ -160,7 +160,8 @@ impl<
             let comp_y = intersect_across(a, b, axis_y);
             // line that tries to intersect both
             let line = Line {
-                origin: comp_x + comp_y,
+                // TODO(review): I don't understand the meaning of origin here
```

Comment:
so, `comp_x` is a point on the `axis_x` axis that is supposed to be between the polygons, and `comp_y` is the same for `axis_y`. Adding them is like adding X and Y projections of a vector.
What the code needs is a line that intersects both polygons, simple as that. Taking `origin = comp_x + comp_y` is not entirely correct here, it's a rough approximation (hence, the TODO at the top).